lang/cs/Org.Apache.REEF.Client/YARN/RESTClient/YarnClient.cs

```diff
@@ -103,6 +105,33 @@ internal YarnClient(IUrlProvider yarnRmUrlProviderUri, IRestRequestExecutor rest
                     GenerateUrlAndExecuteRequestAsync<NewApplication>(request, cancellationToken);
         }
 
+        public async Task<Application> SubmitApplicationAsync(
+            SubmitApplication submitApplication,
+            CancellationToken cancellationToken)
+        {
+            await new RemoveSynchronizationContextAwaiter();
+
+            var request = new RestRequest
+            {
+                Resource = _baseResourceString + SubmitApplication.Resource,
+                Method = Method.POST,
+                RequestFormat = DataFormat.Json,
+                JsonSerializer = new RestJsonSerializer()
+            };
+
+            request.AddBody(submitApplication);
+            var submitResponse = await GenerateUrlAndExecuteRequestAsync(request, cancellationToken);
+
+            if (submitResponse.StatusCode != HttpStatusCode.Accepted)
```

Comment:
This is a quirk in the way the Yarn Rest API is defined. The StatusCode checking in RestRequestExecutor is general purpose error handling.

The StatusCode checking in SubmitApplication in YarnClient is API definition, hence the special treatment.
From http://hadoop.apache.org/docs/r2.6.0/hadoop-yarn/hadoop-yarn-site/ResourceManagerRest.html#Cluster_Applications_APISubmit_Application

POST Response Examples

POST requests can be used to submit apps to the ResourceManager. As mentioned above, an application-id must be obtained first. Successful submissions result in a 202 response code and a Location header specifying where to get information about the app.